### DISCUSSION TITLE
Google Pay payment doesn't get processed

### DISCUSSION THREAD

**Discussion by rinorzek** [swellstores/community#331] [Help] [answered]
Here is what I have for my instant payment button:
```
import { useEffect, useState } from "react";
import type { FC } from "react";
import swell from "swell-js";

import { useCartContext } from "@/base/app/hooks/use-cart-context";
swell.init(process.env.NEXT_PUBLIC_SWELL_STORE_ID, process.env.NEXT_PUBLIC_SWELL_PUBLIC_KEY, {
  useCamelCase: true,
});

import styles from "./styles.module.css";

export const InstantPaymentMethods: FC = () => {
  const [isLoading, setIsLoading] = useState(true);
  const { cart } = useCartContext();
  //
  useEffect(() => {
    const renderPaymentElements = async () => {
      setIsLoading(true);
      await swell.payment.createElements({
        google: {
          elementId: "googlepay-button",
          require: {
            shipping: true,
            email: true,
            phone: true,
          },

          onSuccess: event => {
            setIsLoading(false);
            console.log("event:", event);
          }, // Optional, called on submit Google Pay modal
          onError: error => {
            setIsLoading(false);

            console.error(error);
          }, // Optional, called on payment error
        },
      });
    };

    renderPaymentElements();
  }, []);
  //
  return (
    <div>
      <div className={styles["button-group"]}>
        <div id="googlepay-button"></div>
        <div className={styles["apple-pay-placeholde"]} />
      </div>
    </div>
  );
};

```

And after I click on the Google Pay button, it opens the popup and I click continue (to pay) but after that it just closes and nothing happens, it looks like the onSuccess function is triggered, but when I check Swell orders, that is not processed, what is the problem, can someone help me with this? Thanks.

**Comment by rinorzek** [comment]
One more thing, I found out that those "orders" are at least getting added in the "carts" in the Swell dashboard, and I found this https://github.com/orgs/swellstores/discussions/116, but can someone help me what this means: "In the case for Apple or Google Pay, you will see these appear as credit cards in the Swell dashboard." because as I said I'm not getting anything from the onSuccess callback.

**Reply by vstudenichnik-insoft** [reply, marked as answer]
Hey @rinorzek, the `onSuccess` callback has no arguments. It is called upon successful processing of payment data from the Google Pay modal. This means that the cart billing is configured correctly and the cart is ready to be converted into an order (`swell.cart.submitOrder()`).

**Reply by rinorzek** [reply]
@vstudenichnik-insoft thanks a lot that makes sense, got it working now. Btw have you ever encountered this https://github.com/orgs/swellstores/discussions/333
